packages/next/src/telemetry/events/build.ts

```diff
@@ -212,3 +213,21 @@ export function eventPackageUsedInGetServerSideProps(
     },
   }))
 }
+
+export const ERROR_THROWN_EVENT = 'NEXT_ERROR_THROWN'
+type ErrorThrownEvent = {
+  eventName: typeof ERROR_THROWN_EVENT
+  payload: {
+    errorCode: string | undefined
+  }
+}
+
+// Creates a Telemetry event for errors. For privacy, only includes the error code.
+export function eventErrorThrown(error: Error): ErrorThrownEvent {
+  return {
+    eventName: ERROR_THROWN_EVENT,
+    payload: {
+      errorCode: extractNextErrorCode(error) ?? error.name,
```

Comment:
Can `error` be something that's thrown from outside Next.js? One could argue that `error.name` is leaking because it could be custom info. What do you think about falling back to "ExternalError" instead?

Reply:
We could do that, but unfortunately the `error.name` case is actually what `TurbopackInternalError` is using right now -- I'm guessing this is because their construction is dynamic and can't be analyzed by the error code script. Is it all right for me to manually set an error code?